File: 0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s.length() < 2) return false;
        unordered_map<char, char> hash{
            {'{','}'}, {'(',')'}, {'[',']'}
        };
        stack<char> r;

        for(auto i : s){
            if(i == '{' || i == '(' || i == '['){
                r.push(i);
            }
            else{
                if(!r.empty()){
                    if(hash[r.top()] == i){
                        r.pop();
                    }
                    else return false;
               }
               else return false;
            }
        }
        return r.size() == 0? true : false;
    }
};

File: 0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        map = {')':'(', '}':'{', ']':'['}
        
        for _, key in enumerate(s):
            if key in map.values():
                stack.append(key)

            elif stack and key in map and map[key] == stack[-1]:
                del stack[-1]            

            else: return False

        return True if not stack else False
